src/semgrep_agent/semgrep.py

```diff
@@ -153,25 +136,24 @@ def invoke_semgrep(
             args = ["--skip-unknown-extensions", "--json", *config_args]
             for path in chunk:
                 args.append(path)
-            count = 0
             for result in json.loads(str(semgrep(*args)))["results"]:
-                _update_finding_set(result, committed_datetime, findingSet.current_map)
-                count += 1
+                finding = Finding.from_semgrep_result(result, committed_datetime)
+                while finding in findings.current:
```

Comment:
I think this won't work under the following test plan:

Commit:
```py
5 == 5
```

Then alter file to
```py
6 == 6

5 == 5
```

You should only see 1 new finding in this case.

Reply:
FWIW I ran this test plan on the existing code, and it works as it should currently.

Reply:
OK, thinking about this a bit more, I think ^^ will pass.

However, I see issues with the following:

Commit a new file with two findings in it:
```
2 == 2

2 == 2
```
The agent will only send one finding to the backend, and it will have the hash with index = 1.

Then if we fix one of the findings:
```
2 == 2
```
The backend will detect a new finding (!), because it now has the hash with index 0, for the first time.

Reply:
Also just tested this, and your code works. Took me a while to understand why, so maybe that's a flag that a comment is needed to explain your algorithm.

But your solution is definitely simpler than mine, so let's use it.

Also FYI, @brendongo expressed a preference for putting the finding-set mutation logic inside the FindingsSet object, so may want him to check out this code too.

Reply:
Yes if possible it would be nice if users of the FindingsSet object don't need to know about indexes and having to evolve finding objects. Can probably have FindingSets expose an add_to_baseline/current method